### ISSUE TITLE
Migrate to JIRA Issue Tracker

### ISSUE THREAD

**Issue by varl** [dhis2/notes#135]
The discussion about "GitHub Issues vs. JIRA" is unlikely to have
slipped by unnoticed. It is a topic that is controversial with strong
opinions and good arguments, for and against.

The general consensus is that since JIRA is the official issue tracker
for DHIS2, we can better visualize dependencies across the entire
project if we move everything into that single tool. That said, I am
not going to reiterate all of the arguments here and instead focus
on the practical side of things.

**How do we _actually_ move from GitHub Issues to JIRA?**

First we need to lay out the current structure.

## Current JIRA project structure:

-   [DHIS2](https://jira.dhis2.org/plugins/servlet/project-config/DHIS2/summary)
    -   The project for the Core and all apps.
    -   Organised by components.
        -   Each part of the backend has a "[api]" component.
        -   Each web application has a "[app]" component.
    -   A release spans across the entire suite of issues.
        -   This works because all apps are bundled so their independent
            versions are not tracked.

-   [TECH](https://jira.dhis2.org/plugins/servlet/project-config/TECH/summary)
    -   Internal project for all the "technical" work.
    -   Organised by components (out of necessity).
        - Simplified by using "Frontend" and "Backend" components
          instead of replicating all available in the DHIS2 project.

- [HUB](https://jira.dhis2.org/plugins/servlet/project-config/HUB/summary)
    -    Manages the App Hub server and client.

When we move to continuous application versions, then the web
applications and core will no longer have the same release cycle and
apps will need to manage their own versions.

Further than that, each web application depends on a suite of libraries
that we develop, which in turn has their own versions and releases.

Today, our libraries are managed independently completely outside of
JIRA, while our applications are managed in the context of the DHIS2
project in JIRA.

## Future JIRA project structure

-   [DHIS2](https://jira.dhis2.org/plugins/servlet/project-config/DHIS2/summary)
    -   Manages the Core.
    -   Only has components related to the backend.
    -   The release versions only relate to the Core version.

-   [APPS](https://jira.dhis2.org/plugins/servlet/project-config/APPS/summary)
    -   Manages Web Applications.
    -   Has components per application.

-   [LIBS](https://jira.dhis2.org/plugins/servlet/project-config/LIBS/summary)
    -   Manages Web Libraries.
    -   Has components per library.

-   [TECH](https://jira.dhis2.org/plugins/servlet/project-config/LIBS/summary)
    -   _DHIS2 Internal_
    -   Manages internal technical work.
    -   No changes from today.

- [HUB](https://jira.dhis2.org/plugins/servlet/project-config/HUB/summary)
    -    Manages the App Hub server and client.

- [CLI](https://jira.dhis2.org/plugins/servlet/project-config/CLI/summary)
    -   Manages D2 CLI.
    -   Has components per module.

## Phase 1: Migrate libraries

0.  Set up JIRA project for LIBS
0.  Move GitHub Issues to JIRA per repo.
0.  No release version tracking in JIRA.
0.  Keep the release pipeline as it is today.
0.  Set up simple workflow in JIRA for LIBS project.
0.  Set up source code links for Smart Commits.
0.  Disable GitHub Issues in repository
0.  Add deeplinks and info on how to report issues for the library in README.

Example:

- https://jira.dhis2.org/browse/LIBS-1
- https://github.com/dhis2/prop-types
- https://github.com/dhis2/prop-types/#report-an-issue

## Phase 2:

...

## Phase 3:

...

**Comment by varl** [comment]
# Migrated

- https://github.com/dhis2/prop-types
- https://github.com/dhis2/d2

**Comment by varl** [comment]
Phase 1 has been completed.

**Comment by varl** [comment]
**tl;dr**: I have re-enabled the issues section and added a `comment-and-close` workflow to the library repositories.

There are tools we use that are deeply integrated into GitHub that create issues in certain situations, for example, if something fails. Dependabot, semantic-release, and Kodiak do this, and there may be more.

Disabling Issues cause these integrations to fail, so we need to keep Issues available to those tools. At the same time, we want to inform everyone that our issue tracker is JIRA, and close any issues that are created by non-bot users.

This has been done for all the migrated repos in the list above.

**Comment by varl** [comment]
I went over the apps as well and added the `comment-and-close` workflow.

Only these apps have issues in them that need to be migrated to JIRA. At least no new issues can be opened by non-exempt user from now on.

- https://github.com/dhis2/capture-app/issues
- https://github.com/dhis2/import-export-app/issues
- https://github.com/dhis2/reports-app/issues
- https://github.com/dhis2/usage-analytics-app/issues
- https://github.com/dhis2/sms-configuration-app/issues

**Comment by varl** [comment]
This is more or less obsolete now that there are no open issues on the repos noted above: https://github.com/dhis2/notes/issues/135#issuecomment-731216466

Phase 2 is moving apps out from the core project to the apps project within JIRA, but it is tracked elsewhere.

Closing this.
